docker/Dockerfile.deps

```diff
@@ -31,53 +31,69 @@ RUN LIBZ_VERSION=1.2.11 && \
       CFLAGS="-fPIC" \
       CXXFLAGS="-fPIC" \
     ./configure --prefix=/usr/local && \
-    make install -j"$(grep ^processor /proc/cpuinfo | wc -l)" && \
+    make -j"$(grep ^processor /proc/cpuinfo | wc -l)" && \
+    make install && \
     cd && \
     rm -rf /tmp/zlib-${LIBZ_VERSION}
 
 # CMake
 RUN CMAKE_VERSION=3.13 && \
     CMAKE_BUILD=3.13.5 && \
+    cd /tmp && \
     curl -L https://cmake.org/files/v${CMAKE_VERSION}/cmake-${CMAKE_BUILD}.tar.gz | tar -xzf - && \
-    cd /cmake-${CMAKE_BUILD} && \
+    cd cmake-${CMAKE_BUILD} && \
     ./bootstrap --parallel=$(grep ^processor /proc/cpuinfo | wc -l) && \
-    make -j"$(grep ^processor /proc/cpuinfo | wc -l)" install && \
-    rm -rf /cmake-${CMAKE_BUILD}
+    make -j"$(grep ^processor /proc/cpuinfo | wc -l)" && \
+    make install && \
+    cd && \
+    rm -rf /tmp/cmake-${CMAKE_BUILD}
 
 # protobuf
 RUN PROTOBUF_VERSION=3.14.0 && \
+    cd /tmp && \
     curl -L https://github.com/google/protobuf/releases/download/v${PROTOBUF_VERSION}/protobuf-all-${PROTOBUF_VERSION}.tar.gz | tar -xzf - && \
-    cd /protobuf-${PROTOBUF_VERSION} && \
+    cd protobuf-${PROTOBUF_VERSION} && \
     ./autogen.sh && \
     ./configure CXXFLAGS="-fPIC" --prefix=/usr/local --disable-shared 2>&1 > /dev/null && \
-    make -j"$(grep ^processor /proc/cpuinfo | wc -l)" install 2>&1 > /dev/null && \
-    rm -rf /protobuf-${PROTOBUF_VERSION}
+    make -j"$(grep ^processor /proc/cpuinfo | wc -l)" 2>&1 > /dev/null && \
+    make install 2>&1 > /dev/null && \
```

Comment:
I see the value of moving to tmp, but why split build and install stages?

Reply:
Now I see

Reply:
As a user I wouldn't like to build as sudo, because all the generated files will be owned by root. I think it is cleaner to first build and then install as sudo.